### PR TITLE
Add .well-known support

### PR DESCRIPTION
**Pull request by deepbluev7-old**
With this nheko autodiscovers the server by looking at `/.well-known/matrix/client`. [Specification](https://matrix.org/docs/spec/client_server/latest#server-discovery).

I've also implemented handling the well_known field in login responses, but I don't verify the url there. I expect servers which have such a setup to have setup it correctly. Otherwise I don't know how to handle it correctly, fail the login and request the server name?

The translation update is quite big, as it seems like the translations weren't kept up to date.

This won't build yet, I guess, because I haven't updated the mtxclient hash. I'll do that, when the changes in https://github.com/Nheko-Reborn/mtxclient/pull/19 land. For now this is just provided to ease testing.

### CODE REVIEW

**Comment by deepbluev7** [comment]
Ups, seems like I missed a case, when handling the login response. It seems to always return the new servername, but terminated with a slash, which mtxclient doesn't seem to handle gracefully.

**Comment by deepbluev7** [comment]
Alright, I fixed the login issue by removing terminating slashes in mtxclient, when setting the server name. With that, I can also login, if my server sends a different server name in the login response.

File: src/LoginPage.cpp [hunk 1]
```diff
@@ -186,7 +187,37 @@ LoginPage::onMatrixIdEntered()
...
+                                          tr("Autodiscovery failed. Received malformed response."));
+                                        nhlog::net()->error(
+                                          "Autodiscovery failed. Received malformed response.");
+                                        return;
+                                }
+
+                                emit versionErrorCb(tr("Autodiscovery failed. Unknown error when "
+                                                       "requesting .well-known."));
+                                nhlog::net()->error("Autodiscovery failed. Unknown error when "
+                                                    "requesting .weel-known.");
```
Comment (z3ntu) [inline src/LoginPage.cpp]: typo

Comment (deepbluev7) [reply src/LoginPage.cpp]: Should be fixed

**Comment by redsky17** [comment]
Once https://github.com/Nheko-Reborn/mtxclient/pull/19 goes in, make sure to update the commit and hash for mtxclient in CMakeLists.txt.  Afterwards, I'll merge this.  Thanks for updating the translations as well!  I have not done a very good job about keeping those up-to-date with changes.

**Pull request merged**